### ISSUE TITLE
Async fixture teardown

### ISSUE THREAD

**Issue by petr-k** [pytest-dev/pytest-asyncio#70]
Hi!
I have question regarding how asynchronous fixture teardown (e.g. closing a connection that was open asynchronously) is supposed to work.

When I have a fixture such as:
```
@pytest.fixture(scope='session')
@async_generator
async def fixture():
    # ... async opening of a connection
    connection = await asyncpg.connect(...)
    await yield_(connection)
    await connection.close()
```

I am getting the following error: `ValueError: Async generator fixture didn't stop.Yield only once.`

**Comment by asvetlov** [comment]
Did you try it with Python 3.6, not Python 3.5 and `async_generator` library?

**Comment by petr-k** [comment]
I can confirm that with Python 3.6, it works:
```python
@pytest.yield_fixture()
async def fixture():
    # ...
    yield abc
    await abc.close()
```

**Comment by dgilge** [comment]
Is there a solution for Python 3.5, too? Because it was nice to be able to confirm that packages work with Python 3.5, too.

**Comment by mikenerone** [comment]
Yeah, why did this get closed? It's still valid for Python 3.5.

**Comment by Gatsby-Lee** [comment]
I used to have code like this
```
@pytest.fixture(scope="function", name="redis_client")
@pytest.mark.asyncio
async def fixture_redis_client(function_scoped_container_getter):

    service = function_scoped_container_getter.get("redis").network_info[0]

    hostname = service.hostname
    port = service.host_port

    redis_client = None
    for _ in range(3):
        try:
            redis_client = await aioredis.create_redis(
                f"redis://{hostname}:{port}/0?encoding=utf-8"
            )
            return redis_client
        except Exception:
            await asyncio.sleep(1)
```

AND, changed this way

```
@pytest.yield_fixture(scope="function", name="redis_client")
async def fixture_redis_client(function_scoped_container_getter):

    service = function_scoped_container_getter.get("redis").network_info[0]

    hostname = service.hostname
    port = service.host_port

    redis_client = None
    for _ in range(3):
        try:
            redis_client = await aioredis.create_redis(
                f"redis://{hostname}:{port}/0?encoding=utf-8"
            )
            yield redis_client
            break
        except Exception:
            await asyncio.sleep(1)

    redis_client.close()
    await redis_client.wait_closed()
```
